generators/pkg/generator/generate.go

```diff
@@ -144,3 +155,163 @@ func GeneratePipeline(github *GitHub) (*v1beta1.Pipeline, error) {
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    p.Namespace,
+			Name:         p.Name,
+			GenerateName: p.Name + "-run-",
+			Labels:       p.Labels,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineRef: &v1beta1.PipelineRef{
+				Name: p.Name,
+			},
+			Params: []v1beta1.Param{
+				{
+					Name: "gitrepositoryurl",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "$(params.gitrepositoryurl)",
+					},
+				},
+				{
+					Name: "gitrevision",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "$(params.gitrevision)",
+					},
+				},
+			},
+			Workspaces: []v1beta1.WorkspaceBinding{
+				{
+					Name: p.Spec.Workspaces[0].Name,
+					VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes: []corev1.PersistentVolumeAccessMode{
+								corev1.ReadWriteOnce,
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"storage": *resource.NewQuantity(1*1024*1024*1024, resource.BinarySI),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// create the triggertemplate
+	tt := &v1alpha1.TriggerTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: p.Namespace,
+			Name:      "github-template",
+			Labels:    p.Labels,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       "TriggerTemplate",
+		},
+		Spec: v1alpha1.TriggerTemplateSpec{
+			Params: []v1alpha1.ParamSpec{
+				{
+					Name:        "gitrevision",
+					Description: "The git revision",
+				},
+				{
+					Name:        "gitrepositoryurl",
+					Description: "The git repository url",
+				},
+			},
+			ResourceTemplates: []v1alpha1.TriggerResourceTemplate{
+				{
+					runtime.RawExtension{Object: pr},
+				},
+			},
+		},
+	}
+
+	// create the triggerbinding
+	tb := &v1alpha1.TriggerBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: p.Namespace,
+			Name:      "github-binding",
+			Labels:    p.Labels,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       "TriggerBinding",
+		},
+		Spec: v1alpha1.TriggerBindingSpec{
+			Params: []v1alpha1.Param{
+				{
+					Name:  "gitrevision",
+					Value: "$(body.head_commit.id)",
+				},
+				{
+					Name:  "gitrepositoryurl",
+					Value: "$(body.repository.url)",
+				},
+			},
+		},
+	}
+
+	// create the eventlistener
+	el := &v1alpha1.EventListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: p.Namespace,
+			Name:      "github-listener",
+			Labels:    p.Labels,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       "EventListener",
+		},
+		Spec: v1alpha1.EventListenerSpec{
+			Triggers: []v1alpha1.EventListenerTrigger{
+				{
+					Name: "github-listener",
```

Comment:
nit: for everything but the event listener, we probably want to go with github-push. This will make it easier to add other event types (e.g. pull request).